### PR TITLE
server: Fix exception while update domain resource count (#3204)

### PR DESCRIPTION
**Pull request by DaanHoogland**
reimplement #3303 on moved file in 4.12
(should add the fix from #3204  to master)

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@ustcweizhou , i redid your PR because it wouldn't fwd-merge.

**Comment by ustcweizhou** [comment]
@DaanHoogland ok. what about cherry-pick ?

**Comment by DaanHoogland** [comment]
@ustcweizhou same conflict, this was quicker

**Comment by rhtyd** [comment]
@DaanHoogland you could have forward merged with this change?

**Comment by rhtyd** [comment]
@DaanHoogland can you also check the Travis failure, is this breaking anything?

**Comment by rhtyd** [comment]
@DaanHoogland @ustcweizhou the build fails due to:
```
/home/travis/build/apache/cloudstack/engine/schema/src/main/java/com/cloud/configuration/dao/ResourceCountDaoImpl.java:270:9: expecting SEMI, found 'try'
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.0.0:check (cloudstack-checkstyle) on project cloud-engine-schema: Failed during checkstyle configuration: Exception was thrown while processing /home/travis/build/apache/cloudstack/engine/schema/src/main/java/com/cloud/configuration/dao/ResourceCountDaoImpl.java: MismatchedTokenException occurred during the analysis of file /home/travis/build/apache/cloudstack/engine/schema/src/main/java/com/cloud/configuration/dao/ResourceCountDaoImpl.java. expecting EOF, found 'pstmt' -> [Help 1]
```
Please hold merging until build is fixed, thanks.

File: engine/schema/src/main/java/com/cloud/configuration/dao/ResourceCountDaoImpl.java [hunk 1]
```diff
@@ -266,18 +266,18 @@ public long countMemoryAllocatedToAccount(long accountId) {
     }
 
     private long executeSqlCountComputingResourcesForAccount(long accountId, String sqlCountComputingResourcesAllocatedToAccount) {
-        try (TransactionLegacy tx = TransactionLegacy.currentTxn()) {
+        TransactionLegacy tx = TransactionLegacy.currentTxn()
```
Comment (ustcweizhou) [inline engine/schema/src/main/java/com/cloud/configuration/dao/ResourceCountDaoImpl.java]: @DaanHoogland  you missed ";" at the end of line 269.

**Comment by DaanHoogland** [comment]
@rhtyd fwd-merge did not work due to the file being moved (out of git bounds)

File: engine/schema/src/main/java/com/cloud/configuration/dao/ResourceCountDaoImpl.java (hunk 1 above)
Comment (DaanHoogland) [reply engine/schema/src/main/java/com/cloud/configuration/dao/ResourceCountDaoImpl.java]: fixed

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
closing, setting ```git config global  merge.renamelimit=5635``` would have done the trick.

**Pull request closed without merging**